### DISCUSSION TITLE
FNIRSI GC-01: Recovery

### DISCUSSION THREAD

**Discussion by yochan06** [Gissio/radpro#65] [Q&A] [answered]
FNIRSI GC-01(CH32F103R8T6)not automatically powered on after dragging the firmware,i press the power button and screen only flashes once and then remains black

**Comment by Gissio** [comment]
What is your operating system? Are you able to see the USB drive now? If you are, you can recover your device with the original firmware: https://github.com/Gissio/radpro/tree/main/docs/devices/FNIRSI%20GC-01/firmware .

You can also flash Rad Pro following these instructions: https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install-stlink.md

**Reply by yochan06** [reply]
operating system is windows10 and able to see the USB drive but after dragging the original firmware it still not working
i also try PC with windows 11 ,after dragging the original firmware it still not working
※I need to press the reset button and then connect to the computer and press the power button on the device to see the USB drive

**Reply by Gissio** [reply, marked as answer]
Try the [alternative installation instructions](https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install-stlink.md).

**Comment by phr0stb1t3** [comment]
I had a similar issue on my Fnirsi GC-01, I would drag the firmware and nothing would happen, I thought maybe it was due to the small clock battery inside that was dead in mine (I recalled it would not retain clock settings in the original firmware). I replaced that and tried again - it did not work. 

I grabbed another usb-c cable and went to an actual windows laptop instead of doing USB pass-through via a Windows 10 VM, and voila it worked as described. 

I dragged the firmware into the drive and the device powered on with the new radpro firmware. Everything is working great so far.

When I was testing on the Windows 10 VM I was only able to see the drive if USB was set to 2.0, with USB3 the drive would not even appear. I would suggest trying a different cable and PC if possible.

**Reply by Montecri** [reply]
Mine only works through a USB hub.
